tests/tests_integration/test_file_integration.py

```diff
@@ -38,79 +40,145 @@ def set_test_parameters() -> ParamTest:
         f"util_integration_file_test_3-{test_id}",
         f"util_integration_file-big-2-{test_id}",
     ]
+    test_parameter.space = "core-dm-test"
     return test_parameter
 
 
-def await_is_uploaded_status(client: CogniteClient, external_id: str):
+def await_is_uploaded_status(
+    client: CogniteClient, external_id: Optional[str] = None, instance_id: Optional[NodeId] = None
+) -> None:
     for _ in range(10):
-        if client.files.retrieve(external_id=external_id).uploaded:
+        if external_id is not None:
+            retrieved = client.files.retrieve(external_id=external_id)
+        elif instance_id is not None:
+            retrieved = client.files.retrieve(instance_id=instance_id)
+        else:
+            raise ValueError("Please provide either external_id or instance_id")
+        if retrieved is not None and retrieved.uploaded:
             return
         time.sleep(1)
 
 
 @pytest.mark.parametrize("functions_runtime", ["true", "false"])
-def test_file_upload_queue(set_upload_test: Tuple[CogniteClient, ParamTest], functions_runtime: str):
+def test_file_upload_queue(set_upload_test: Tuple[CogniteClient, ParamTest], functions_runtime: str) -> None:
     os.environ["COGNITE_FUNCTION_RUNTIME"] = functions_runtime
     client, test_parameter = set_upload_test
     queue = FileUploadQueue(cdf_client=client, overwrite_existing=True, max_queue_size=2)
 
     current_dir = pathlib.Path(__file__).parent.resolve()
 
     # Upload a pair of actual files
+    assert test_parameter.external_ids is not None
+    assert test_parameter.space is not None
     queue.add_to_upload_queue(
-        file_meta=FileMetadata(external_id=test_parameter.external_ids[0], name=test_parameter.external_ids[0]),
+        meta_or_apply=FileMetadata(external_id=test_parameter.external_ids[0], name=test_parameter.external_ids[0]),
         file_name=current_dir.joinpath("test_file_1.txt"),
     )
     queue.add_to_upload_queue(
-        file_meta=FileMetadata(external_id=test_parameter.external_ids[1], name=test_parameter.external_ids[1]),
+        meta_or_apply=FileMetadata(external_id=test_parameter.external_ids[1], name=test_parameter.external_ids[1]),
         file_name=current_dir.joinpath("test_file_2.txt"),
     )
     # Upload the Filemetadata of an empty file without trying to upload the "content"
     queue.add_to_upload_queue(
-        file_meta=FileMetadata(external_id=test_parameter.external_ids[3], name=test_parameter.external_ids[3]),
+        meta_or_apply=FileMetadata(external_id=test_parameter.external_ids[3], name=test_parameter.external_ids[3]),
+        file_name=current_dir.joinpath("empty_file.txt"),
+    )
+
+    queue.add_to_upload_queue(
+        meta_or_apply=CogniteExtractorFileApply(
+            external_id=test_parameter.external_ids[0], name=test_parameter.external_ids[0], space=test_parameter.space
```

Comment:
Don't reuse external IDs in the tests. I realize it works, but it's very confusing, and the cleanup won't work like this. I'm reasonably certain you need to clean up these files in some special way.

Reply:
There's no alternative to external ID here because it is required by the instance. Working on cleaning up the files.

Reply:
Updated now.